### DISCUSSION TITLE
object position nor rotation updating?

### DISCUSSION THREAD

**Discussion by dante-biase** [enable3d/enable3d#208] [Help / Questions]
Can't figure out how to change move an object.

code

```
const ball = physics.add.sphere({radius: 1.25, x: -3, y: 10, z: 5 }, { lambert: { color: 0x2194ce } })
ball.name = "ball" 
ball.body.transform()
ball.body.setPosition(-10 , -10, -10)
ball.body.setRotation(0, 0, 0)
ball.body.refresh()
ball.body.needUpdate = true
```

using `ball.position.set` does not work either?

**Comment by vitalspace** [comment]
Hi, try this. 

```
create() {
    this.ball = this.physics.add.sphere({radius: 1.25, x: 0, y: 0, z: 0 , collisionFlags: 1, mass: 100}, { lambert: { color: 0x2194ce } })
}

update() {
    if(this.ball) {
        this.ball.rotation.y += 0.2;
        this.ball.body.needUpdate = true;
    }
}

```

**Comment by yandeu** [comment]
The body has to be kinematic.

See https://enable3d.io/examples/kinematic-body-orbiting-around-sun.html

**Reply by dante-biase** [reply]
I figured that out yeah. 

A workaround solution is to change the collision flag to kinematic and then move the object. The problem is you cannot change the collision flag back to dynamic afterwards as this effectively cancels the transformation.

I assume you would have to make this change the next frame or some other time later in the execution.

Is there a way to move dynamic bodies? if not, maybe thats a feature to consider adding

**Reply by yandeu** [reply]
You can switch between kinematic and dynamic.
See https://enable3d.io/examples/teleport-a-dynamic-body.html

Of course you can move dynamic bodies by applying some force, for example.

**Reply by luddensdesir** [reply]
Is there a way to disable collision on a kinematic body?

**Reply by yandeu** [reply]
@luddens combine kinematic and ghost flags (2 + 4)
See: https://enable3d.io/docs.html#body-types

**Comment by vitalspace** [comment]
Maybe 

```
create() {
    this.ball = this.physics.add.sphere({radius: 1.25, x: 0, y: 0, z: 0 , collisionFlags: 1, mass: 100}, { lambert: { color: 0x2194ce } })
}

update() {
    if(this.ball) {
        // set body to kinematic.
        this.ball.body.setCollisionFlags(2)
        // this.ball.rotation.y += 0.2;
        this.ball.body.needUpdate = true;
    
        this.ball.body.once.update(() => {
            // set body to dinamic 
            this.ball.body.setCollisionFlags(0)
            // if you don't reset the velocity  and angularVelocity, the object will keep it
            this.ball.body.setVelocity(0,0,0)
            this.ball.body.setAngularVelocity(0,0,0)
        })
    }
}
```
